rust/nasl-builtin-isotime/src/lib.rs

```diff
@@ -0,0 +1,145 @@
+// SPDX-FileCopyrightText: 2023 Greenbone AG
+//
+// SPDX-License-Identifier: GPL-2.0-or-later WITH x11vnc-openssl-exception
+
+//! Defines NASL functions regarding isotime.
+
+use chrono::{Datelike, Months, NaiveDate, NaiveDateTime, TimeDelta};
+use nasl_builtin_utils::{Context, FunctionErrorKind, NaslFunction, Register};
+use nasl_function_proc_macro::nasl_function;
+
+const ISOFORMAT: &str = "yyyymmddThhmmss";
+const READABLEFORMAT: &str = "yyyy-mm-dd hh:mm:ss";
+
+fn parse_isotime(time: &str) -> Option<NaiveDateTime> {
+    NaiveDateTime::parse_from_str(time, "%Y%m%dT%H%M%S").ok()
+}
```

Comment:
Just a comment, no need to do anything:

I think it would be much cleaner if we could do
```rust
impl<'a> FromNaslValue<'a> for NaiveDateTime {
    fn from_nasl_value(value: &'a nasl_syntax::NaslValue) -> Result<Self, FunctionErrorKind> {
        let time_str = <&str>::from_nasl_value(value)?;
        parse_time(time_str)
    }
}
```
and then directly take `NaiveDateTime` in the functions. This has two benefits: It reduces the amount of times we need to call `parse_time` manually, but more importantly, it allows defining some of the NASL functions below as infallible (since they are only fallible due to the parsing of the isotime, which would otherwise be handled by `nasl_function`), which would be quite nice.

However, due to the current crate split situation, this isn't possible because of the orphan rule (`FromNaslValue` is not defined in `nasl-builtin-isotime`), requiring us to do this by defining a wrapper type (like `IsoTime(NaiveDateTime)`) which reduces the ergonomics by quite a lot.

Anyways, feel free to resolve this, because I don't think we can improve this situation in this PR.

Reply:
Would it be worth to resolve the crate split situation and move to a module approach?